### DISCUSSION TITLE
Am I missing the documentation?

### DISCUSSION THREAD

**Discussion by joshuawootonn** [NetTopologySuite/NetTopologySuite#490] [Q&A]
***Full disclosure this is not a issue, it is a question*** 

Hello! I am really new to GIS technology in general, but I am trying to learn! I have looked through this github repo, looked through npgsql.org's docs on spatial databases, looked through your docs (which seem to be more of generated javadoc like information), and looked through references in Github for examples of how to use this library. Personally, I learn best through examples, which is maybe why things haven't _clicked_ for me yet.

I was wondering if someone could point me in the right direction of something like this. Even if it is just a exemplary project. Bonus points if it uses Dapper!!

**Comment by bricelam** [comment]
https://gist.github.com/bricelam/7eca234674c3ca4150872f899af37611

**Comment by juliusfriedman** [comment]
See also https://www.markopapic.com/finding-nearby-users-using-ef-core-spatial-data/

**Comment by richardthombs** [comment]
Yeah, if you are looking for a simple way of storing and working with lat/lng points in an EF Core / ASP.NET app, good luck. I've just managed to get Postgres and EF Core set up and got a point stored. When I looked at what the point looked like when serialised to JSON, check out the insanity that it produced. All I want is a simple lat/lng point!

I'm ditching it and going back to simple lat and lng columns until I can see a reason to dig into this further.

```
"point": {
"coordinateSequence": {
"dimension": 2,
"measures": 0,
"spatial": 2,
"ordinates": "xy",
"hasZ": false,
"hasM": false,
"zOrdinateIndex": -1,
"mOrdinateIndex": -1,
"count": 1
},
"coordinates": [
{
"x": -1.6918626,
"y": 55.0024163,
"z": "NaN",
"m": "NaN"
}
],
"numPoints": 1,
"isEmpty": false,
"dimension": "point",
"boundaryDimension": "false",
"x": -1.6918626,
"y": 55.0024163,
"coordinate": {
"x": -1.6918626,
"y": 55.0024163,
"z": "NaN",
"m": "NaN"
},
"geometryType": "Point",
"ogcGeometryType": "point",
"boundary": [],
"z": "NaN",
"m": "NaN",
"factory": {
"precisionModel": {
"isFloating": true,
"maximumSignificantDigits": 16,
"scale": 0,
"precisionModelType": "floating"
},
"coordinateSequenceFactory": {
"ordinates": "xyzm"
},
"srid": 4326
},
"userData": null,
"srid": 4326,
"precisionModel": {
"isFloating": true,
"maximumSignificantDigits": 16,
"scale": 0,
"precisionModelType": "floating"
},
"numGeometries": 1,
"isSimple": true,
"isValid": true,
"area": 0,
"length": 0,
"envelopeInternal": {
"isNull": false,
"width": 0,
"height": 0,
"minX": -1.6918626,
"maxX": -1.6918626,
"minY": 55.0024163,
"maxY": 55.0024163,
"area": 0,
"minExtent": 0,
"maxExtent": 0,
"centre": {
"x": -1.6918626,
"y": 55.0024163,
"z": "NaN",
"m": "NaN"
}
},
"isRectangle": false
}
```

**Comment by airbreather** [comment]
> I've just managed to get Postgres and EF Core set up and got a point stored. When I looked at what the point looked like when serialised to JSON, check out the insanity that it produced. All I want is a simple lat/lng point!

Adding the converters from the NetTopologySuite.IO.GeoJSON package to the serializer settings should make it write out as [GeoJSON](https://geojson.org/).

Here's the wiki entry on how to enable it: https://github.com/NetTopologySuite/NetTopologySuite.IO.GeoJSON/wiki/HowTo-...#use-it-with-aspnet-core-v3

**Reply by douglasg14b** [reply]
Non broken link: https://github.com/NetTopologySuite/NetTopologySuite.IO.GeoJSON/wiki/Using-NTS.IO.GeoJSON-with-ASP.NET-Core-MVC#aspnet-core-mvc-30-or-higher

**Comment by richardthombs** [comment]
Thanks, I did read that, but I am trying to get to something like ```point: {lat: N, lng: N}```.

**Comment by joshuawootonn** [comment]
This issue is related to Dapper not EF. I don't have the context to comment on what @airbreather is talking about, but if you are just trying to write a point to the database I would just use Dapper and skip the ORM. 🤷‍♀️

**Comment by joshuawootonn** [comment]
I wouldn't say it is viewer friendly, but you can look at how I am writing CRUDing a LineString here: https://github.com/joshuawootonn/42colors/tree/master/backend

and then there is more in this branch/file, which includes getting LineString within a Polygon
https://github.com/joshuawootonn/42colors/blob/feature/lines-by-mapPosition/backend/backend.core/Repositories/LineRepository.cs
